scripts/corelli/calibration/utils.py

```diff
@@ -0,0 +1,56 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+import numpy as np
+from typing import Union
+
+# imports from Mantid
+from mantid.dataobjects import TableWorkspace, Workspace2D
+from Calibration import tube
+from Calibration.tube_calib_fit_params import TubeCalibFitParams
+
+InputWorkspace = Union[str, Workspace2D]  # type alias
+
+
+def calculate_tube_calibration(workspace: InputWorkspace, tube_name: str, shadow_height: float = 1000,
+                               shadow_width: float = 4, fit_domain: float = 7) -> TableWorkspace:
+    r"""
+    Calibration table for one tube of CORELLI
+
+    :param workspace: string or handle to ~mantid.dataobjects.Workspace2D
+    :param tube_name: string uniquely representing one tube e.g. 'bank88/sixteenpack/tube3'
+    :param shadow_height: estimated dip in the background intensity.
+    :param shadow_width: estimated width of the shadow cast by the wire, in pixel units
+    :param fit_domain: estimated range, in pixel units, over which to carry out the fit.
+
+    :return: table containing detector ID and position vector
+    """
+    message = f'Cannot process workspace {workspace}. Pass the name of an existing workspace or a workspace handle'
+    assert isinstance(workspace, (str, Workspace2D)), message
+    assert shadow_height > 0, 'shadow height must be positive'
+    for marker in ('bank', 'sixteenpack', 'tube'):
+        assert marker in tube_name, f'{tube_name} does not uniquely specify one tube'
+    peak_height, peak_width = -shadow_height, shadow_width
+    wire_gap = (2 * 25.4 + 2) / 1000  # gap along the Y-coordinate between consecutive wire centers
+    # the center of the 16 wires is aligned with the center of the tube, set to Y == 0
+    wire_positions = np.arange(-7.5 * wire_gap, 8.5 * wire_gap, wire_gap)
```

Comment:
Some explanation of the hard coded constants will be very helpful.

Reply:
I added constants `WIRE_GAP`, `bottom_wire_position`, and `topmost_wire_position`